shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/util/TableMetaDataUtil.java

```diff
@@ -55,4 +58,24 @@
         }
         return result;
     }
+    
+    /**
+     * Get data source all actual table groups.
+     *
+     * @param tableNames table name collection
+     * @param materials materials
+     * @return datasource and table collection map
+     */
+    public static Map<String, Collection<String>> getDataSourceAllActualTableGroups(final Collection<String> tableNames, final SchemaBuilderMaterials materials) {
```

Comment:
@tuichenchuxin You may consider publishing a community task to add test cases for TableMetaDataUtil.